### ISSUE TITLE
Run-AlValidation throws "container name already in use error"

### ISSUE THREAD

**Issue by buzzwick** [microsoft/navcontainerhelper#3092]
Running Run-ALValidation completes, but Validation Results and Run-ALValidation Failure come up with 

`Unexpected error while validating app. Error is: docker: Error response from daemon: Conflict. The container name "/bcvalidation" is already in use by container "3dbf34fe4a3f7c81a5d41a4161a08d708da49eb2daf33d6180468f9592199d67". You have to remove (or rename) that container to be able to reuse that name.`
 
I have tried with the default containername "bcserver", and with an explicit name, "bcvalidation." We are validating against three locales--US, DE, GB--and Run-AlValidation seems to be creating a container for each locale. Relevant output reproduced below. 

Run script:

```
$validationResults = Run-AlValidation 
    -validateCurrent 
    -apps @( "C:\Users\dbaldwin\OneDrive - Computer Associates, Inc. - MBS\Documents\_ALPROJECTS\CAI-Software-Precious-Metals-Solution\CAI Software Precious Metals Solution\CAI Software_CAI Software Precious Metals Solution_21.9.0.0.app" ) 
    -countries @( "US","DE","GB" ) 
    -affixes @( "CAI" ) 
    -supportedCountries @( "US", "DE", "GB" ) 
    -containerName bcvalidation
$validationResults | Write-Host -ForegroundColor Red
```

```
C:\ProgramData\BcContainerHelper\a5a8e828-eaba-4b57-aa60-e5dc93d3e9c2\CAI Software_CAI Software Precious Metals Solution_21.9.0.0.app successfully created in 16 seconds
Ignoring warnings
Ignoring infos
Copy C:\ProgramData\BcContainerHelper\8111d71a-44ca-4852-8c87-25f69c45cdef\CAI Software_CAI Software Precious Metals Solution_21.9.0.0.app to C:\ProgramData\BcContainerHelper\0467936f-43a2-4fd3-b2da-0d4cf8229764

Running AppSourceCop took 32 seconds

   _____                _   _                               _        _
  / ____|              | | (_)                             | |      (_)
 | |     _ __ ___  __ _| |_ _ _ __   __ _    ___ ___  _ __ | |_ __ _ _ _ __   ___ _ __
 | |    | '__/ _ \/ _` | __| | '_ \ / _` |  / __/ _ \| '_ \| __/ _` | | '_ \ / _ \ '__|
 | |____| | |  __/ (_| | |_| | | | | (_| | | (__ (_) | | | | |_ (_| | | | | |  __/ |
  \_____|_|  \___|\__,_|\__|_|_| |_|\__, |  \___\___/|_| |_|\__\__,_|_|_| |_|\___|_|
                                     __/ |
                                    |___/

Creating container for country de
BcContainerHelper is version 4.0.14
BcContainerHelper is running as administrator
HyperV is Enabled
UsePsSession is True
Host is Microsoft Windows 10 Enterprise - 10.0.19045.3031
Docker Client Version is 24.0.2
Docker Server Version is 24.0.2
WARNING: Ignoring custom format, because both --format and --quiet are set.
Removing entries from hosts
Removing bcvalidation from container hosts file
Removing bcvalidation-* from container hosts file
Removing Desktop shortcuts
Removing C:\ProgramData\BcContainerHelper\Extensions\bcvalidation
Fetching all docker images
Fetching all docker volumes
Using image mcr.microsoft.com/businesscentral:10.0.19041.1415-filesonly
Creating Container bcvalidation
Style: sandbox
Multitenant: Yes
Version: 22.2.56969.57908
Platform: 22.0.57885.0
Generic Tag: 1.0.2.14
Container OS Version: 10.0.19041.1415 (2004)
Host OS Version: 10.0.19045.3031 (22H2)
WARNING: Host OS is Windows 10 21H1 or newer and Container OS is 2004, defaulting to process isolation. If you experience problems, add -isolation hyperv.
Using process isolation
Using locale de-DE
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Downloading C:\ProgramData\BcContainerHelper\Extensions\bcvalidation\Microsoft.VisualStudio.Services.VSIXPackage.vsix
Using HttpClient
Additional Parameters:
--env customNavSettings=EnableTaskScheduler=True
Files in C:\ProgramData\BcContainerHelper\Extensions\bcvalidation\my:
- AdditionalOutput.ps1
- MainLoop.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container bcvalidation from image mcr.microsoft.com/businesscentral:10.0.19041.1415-filesonly
New-BcContainer Telemetry Correlation Id: 184eed99-ab36-4b1d-ad7b-30236cb2926b
Unexpected error while validating app. Error is: docker: Error response from daemon: Conflict. The container name "/bcvalidation" is already in use by container "3dbf34fe4a3f7c81a5d41a4161a08d708da49eb2daf33d6180468f9592199d67". You have to remove (or rename) that container to be able to reuse that name.
See 'docker run --help'.
ExitCode: 125
Commandline: docker run --volume "c:\bcartifacts.cache:c:\dl" --label nav= --env isBcSandbox=Y --label version=22.2.56969.57908 --label platform=22.0.57885.0 --label country=DE --env artifactUrl=https://bcartifacts.azureedge.net/sandbox/22.2.56969.57908/de --env multitenant=Y --env licenseFile="" --name bcvalidation --hostname bcvalidation --env auth=NavUserPassword --env username="admin" --env ExitOnError=N --env locale=de-DE --env databaseServer="" --env databaseInstance="" --volume "C:\ProgramData\BcContainerHelper:C:\ProgramData\BcContainerHelper" --volume "C:\ProgramData\BcContainerHelper\Extensions\bcvalidation\my:C:\Run\my" --isolation process --restart unless-stopped --env 
filesOnly=True --memory 8G --env enableApiServices=Y --env useSSL=N --volume "c:\windows\system32\drivers\etc:C:\driversetc" --env securePassword=76492d1116743f0423413b16050a5345MgB8AGQAbQA2AE0ASwBHAHgAbwBaAFEAbQBRAHIAWQBOAG4AegB5ADYAaAByAHcAPQA9AHwAYwA5AGQAYQBjAGYAYgBmADAAYQBiADkANQBhAGQAZQA5ADIANwBhAGIAZQBmADgAMwA1AGMAMQA5ADMANgBmADAANgA5ADQAOQBkADQAMgBjAGUAYQBkAGUANwAxADIANQBiADUAZQBmADYAZQAxADgAYQA2ADUAMwA2AGMANQA= --env passwordKeyFile="c:\run\my\aes.key" --env removePasswordKeyFile=Y --env customNavSettings=EnableTaskScheduler=True --env accept_eula=Y --env accept_outdated=Y --detach mcr.microsoft.com/businesscentral:10.0.19041.1415-filesonly
at DockerDo, C:\Users\dbaldwin\OneDrive - Computer Associates, Inc. - MBS\Documents\PowerShell\Modules\BcContainerHelper\4.0.14\HelperFunctions.ps1: line 234
at New-BcContainer, C:\Users\dbaldwin\OneDrive - Computer Associates, Inc. - MBS\Documents\PowerShell\Modules\BcContainerHelper\4.0.14\ContainerHandling\New-NavContainer.ps1: line 1949
at <ScriptBlock>, C:\Users\dbaldwin\OneDrive - Computer Associates, Inc. - MBS\Documents\PowerShell\Modules\BcContainerHelper\4.0.14\AppHandling\Run-AlValidation.ps1: line 320
at <ScriptBlock>, C:\Users\dbaldwin\OneDrive - Computer Associates, Inc. - MBS\Documents\PowerShell\Modules\BcContainerHelper\4.0.14\AppHandling\Run-AlValidation.ps1: line 466
at <ScriptBlock>, C:\Users\dbaldwin\OneDrive - Computer Associates, Inc. - MBS\Documents\PowerShell\Modules\BcContainerHelper\4.0.14\AppHandling\Run-AlValidation.ps1: line 445
at <ScriptBlock>, C:\Users\dbaldwin\OneDrive - Computer Associates, Inc. - MBS\Documents\PowerShell\Modules\BcContainerHelper\4.0.14\AppHandling\Run-AlValidation.ps1: line 429
at <ScriptBlock>, C:\Users\dbaldwin\OneDrive - Computer Associates, Inc. - MBS\Documents\PowerShell\Modules\BcContainerHelper\4.0.14\AppHandling\Run-AlValidation.ps1: line 393
at Run-AlValidation, C:\Users\dbaldwin\OneDrive - Computer Associates, Inc. - MBS\Documents\PowerShell\Modules\BcContainerHelper\4.0.14\AppHandling\Run-AlValidation.ps1: line 391
at <ScriptBlock>, C:\Users\dbaldwin\OneDrive - Computer Associates, Inc. - MBS\Documents\_ALPROJECTS\_PS\Docker\Run-ALValidation.ps1: line 1
at <ScriptBlock>, <No file>: line 1

  _____                           _                _____            _        _
 |  __ \                         (_)              / ____|          | |      (_)
 | |__) |___ _ __ ___   _____   ___ _ __   __ _  | |     ___  _ __ | |_ __ _ _ _ __   ___ _ __
 |  _  // _ \ '_ ` _ \ / _ \ \ / / | '_ \ / _` | | |    / _ \| '_ \| __/ _` | | '_ \ / _ \ '__|
 | | \ \  __/ | | | | | (_) \ V /| | | | | (_| | | |____ (_) | | | | |_ (_| | | | | |  __/ |
 |_|  \_\___|_| |_| |_|\___/ \_/ |_|_| |_|\__, |  \_____\___/|_| |_|\__\__,_|_|_| |_|\___|_|
                                           __/ |
                                          |___/

WARNING: Ignoring custom format, because both --format and --quiet are set.
Removing entries from hosts
Removing bcvalidation from container hosts file
Removing bcvalidation-* from container hosts file
Removing Desktop shortcuts
Removing C:\ProgramData\BcContainerHelper\Extensions\bcvalidation

Removing container took 0 seconds

AL Validation finished in 125 seconds
 __      __   _ _     _       _   _               _____                _ _       
 \ \    / /  | (_)   | |     | | (_)             |  __ \              | | |
  \ \  / /_ _| |_  __| | __ _| |_ _  ___  _ __   | |__) |___ ___ _   _| | |_ ___
   \ \/ / _` | | |/ _` |/ _` | __| |/ _ \| '_ \  |  _  // _ \ __| | | | | __/ __|
    \  / (_| | | | (_| | (_| | |_| | (_) | | | | | | \ \  __\__ \ |_| | | |_\__ \
     \/ \__,_|_|_|\__,_|\__,_|\__|_|\___/|_| |_| |_|  \_\___|___/\__,_|_|\__|___/

Unexpected error while validating app. Error is: docker: Error response from daemon: Conflict. The container name "/bcvalidation" is already in use by container "3dbf34fe4a3f7c81a5d41a4161a08d708da49eb2daf33d6180468f9592199d67". You have to remove (or rename) that container to be able to reuse that name.
See 'docker run --help'.
ExitCode: 125
Commandline: docker run --volume "c:\bcartifacts.cache:c:\dl" --label nav= --env isBcSandbox=Y --label version=22.2.56969.57908 --label platform=22.0.57885.0 --label country=DE --env artifactUrl=https://bcartifacts.azureedge.net/sandbox/22.2.56969.57908/de --env multitenant=Y --env licenseFile="" --name bcvalidation --hostname bcvalidation --env auth=NavUserPassword --env username="admin" --env ExitOnError=N --env locale=de-DE --env databaseServer="" --env databaseInstance="" --volume "C:\ProgramData\BcContainerHelper:C:\ProgramData\BcContainerHelper" --volume "C:\ProgramData\BcContainerHelper\Extensions\bcvalidation\my:C:\Run\my" --isolation process --restart unless-stopped --env 
filesOnly=True --memory 8G --env enableApiServices=Y --env useSSL=N --volume "c:\windows\system32\drivers\etc:C:\driversetc" --env securePassword=76492d1116743f0423413b16050a5345MgB8AGQAbQA2AE0ASwBHAHgAbwBaAFEAbQBRAHIAWQBOAG4AegB5ADYAaAByAHcAPQA9AHwAYwA5AGQAYQBjAGYAYgBmADAAYQBiADkANQBhAGQAZQA5ADIANwBhAGIAZQBmADgAMwA1AGMAMQA5ADMANgBmADAANgA5ADQAOQBkADQAMgBjAGUAYQBkAGUANwAxADIANQBiADUAZQBmADYAZQAxADgAYQA2ADUAMwA2AGMANQA= --env passwordKeyFile="c:\run\my\aes.key" --env removePasswordKeyFile=Y --env customNavSettings=EnableTaskScheduler=True --env accept_eula=Y --env accept_outdated=Y --detach mcr.microsoft.com/businesscentral:10.0.19041.1415-filesonly
  _____                          ___      __   _ _     _       _   _               ______    _ _
 |  __ \                   /\   | \ \    / /  | (_)   | |     | | (_)             |  ____|  (_) |
 | |__) |   _ _ __ ______ /  \  | |\ \  / /_ _| |_  __| | __ _| |_ _  ___  _ __   | |__ __ _ _| |_   _ _ __ ___    
 |  _  / | | | '_ \______/ /\ \ | | \ \/ / _` | | |/ _` |/ _` | __| |/ _ \| '_ \  |  __/ _` | | | | | | '__/ _ \   
 | | \ \ |_| | | | |    / ____ \| |  \  / (_| | | | (_| | (_| | |_| | (_) | | | | | | | (_| | | | |_| | | |  __/   
 |_|  \_\__,_|_| |_|   /_/    \_\_|   \/ \__,_|_|_|\__,_|\__,_|\__|_|\___/|_| |_| |_|  \__,_|_|_|\__,_|_|  \___|   

Unexpected error while validating app. Error is: docker: Error response from daemon: Conflict. The container name "/bcvalidation" is already in use by container "3dbf34fe4a3f7c81a5d41a4161a08d708da49eb2daf33d6180468f9592199d67". You have to remove (or rename) that container to be able to reuse that name.
See 'docker run --help'.
ExitCode: 125
Commandline: docker run --volume "c:\bcartifacts.cache:c:\dl" --label nav= --env isBcSandbox=Y --label version=22.2.56969.57908 --label platform=22.0.57885.0 --label country=DE --env artifactUrl=https://bcartifacts.azureedge.net/sandbox/22.2.56969.57908/de --env multitenant=Y --env licenseFile="" --name bcvalidation --hostname bcvalidation --env auth=NavUserPassword --env username="admin" --env ExitOnError=N --env locale=de-DE --env databaseServer="" --env databaseInstance="" --volume "C:\ProgramData\BcContainerHelper:C:\ProgramData\BcContainerHelper" --volume "C:\ProgramData\BcContainerHelper\Extensions\bcvalidation\my:C:\Run\my" --isolation process --restart unless-stopped --env 
filesOnly=True --memory 8G --env enableApiServices=Y --env useSSL=N --volume "c:\windows\system32\drivers\etc:C:\driversetc" --env securePassword=76492d1116743f0423413b16050a5345MgB8AGQAbQA2AE0ASwBHAHgAbwBaAFEAbQBRAHIAWQBOAG4AegB5ADYAaAByAHcAPQA9AHwAYwA5AGQAYQBjAGYAYgBmADAAYQBiADkANQBhAGQAZQA5ADIANwBhAGIAZQBmADgAMwA1AGMAMQA5ADMANgBmADAANgA5ADQAOQBkADQAMgBjAGUAYQBkAGUANwAxADIANQBiADUAZQBmADYAZQAxADgAYQA2ADUAMwA2AGMANQA= --env passwordKeyFile="c:\run\my\aes.key" --env removePasswordKeyFile=Y --env customNavSettings=EnableTaskScheduler=True --env accept_eula=Y --env accept_outdated=Y --detach mcr.microsoft.com/businesscentral:10.0.19041.1415-filesonly

```

**Comment by freddydk** [comment]
and what happens if you remove the containerName parameter?

**Comment by buzzwick** [comment]
I get the same result either way--with or without the containername parameter. 

For what it's worth, I started out with no docker containers, and now I have two: "bcserver" and "bcvalidation."  Both were created by the validation routine. I also tried removing the  specification of supported locales from the app.json file (because it seems to create a new container for each locale), but I get the same result.
